### PR TITLE
(#1304) Implemented RtUser.markAsRead(Date)

### PR DESCRIPTION
**Pull request by llorllale**
This PR:
* Solves #1304 
* Uses [this documentation](https://developer.github.com/v3/activity/notifications/#mark-as-read) as reference
* Leaves puzzle for more work that will then enable the tests included in this PR

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1391](https://github.com/jcabi/jcabi-github/issues/1391) is now in scope, role is `REV`

**Comment by 0crat** [comment]
@amihaiemil[/z](https://www.0crat.com/u/amihaiemil) everybody who has role `REV` are banned at this job; I won't be able to assign anyone automatically; consider assigning someone manually or invite more people to the project, as explained in [§51](http://www.zerocracy.com/policy.html#51)

**Comment by 0crat** [comment]
This pull request [#1391](https://github.com/jcabi/jcabi-github/issues/1391) is assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), here is [why](https://www.0crat.com/footprint/C9R53K5JA/1091810478); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @amihaiemil[/z](https://www.0crat.com/u/amihaiemil) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer

File: src/test/java/com/jcabi/github/RtUserTest.java [hunk 1]
```diff
@@ -475,6 +478,37 @@ public void notifications() throws Exception {
         );
     }
 
+    /**
+     * markAsRead() should complete successfully if response code is 205.
+     * @throws Exception Thrown in case of error.
+     * @todo #1304:30min Un-ignore this test and
```
Comment (amihaiemil) [inline src/test/java/com/jcabi/github/RtUserTest.java]: @llorllale I don't think it is the right direction. The Mk* classes are not made for us necessarily, but for the users of the library, who want to substitute the real api with the mock version... so we shouldn't alter it in such a way that it fits our tests, I believe (but maybe I'm wrong). 

Instead, see how other unit tests are written, they start an in-memory HTTP server (with MkGrizzly, from jcabi-http), instruct it what responses to give to any URI and then you can pass a custom ``Request`` to ``RtGithub`` (it is very similar to what we do in docker-java-api). 

So, I would say change this puzzle to what I said above (write tests with MkGrizzly mock server). WDYT?

File: src/main/java/com/jcabi/github/RtUser.java [hunk 2]
```diff
@@ -158,7 +160,16 @@ public UserEmails emails() {
 
     @Override
     public void markAsRead(final Date lastread) throws IOException {
```
Comment (amihaiemil) [inline src/main/java/com/jcabi/github/RtUser.java]: @llorllale looks nice. Can you leave here a puzzle to start with ``User.notifications()`` as you suggested [here](https://github.com/jcabi/jcabi-github/issues/1304#issuecomment-378642497)? Sorry, by the way, I completely forgot about it, even after I'd seen your ping :(

**Comment by 0crat** [comment]
Manual assignment of issues is discouraged, see [§19](http://www.zerocracy.com/policy.html#19): -5 point(s) just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

File: src/test/java/com/jcabi/github/RtUserTest.java (hunk 1 above)
Comment (llorllale) [reply src/test/java/com/jcabi/github/RtUserTest.java]: @amihaiemil ok

**Comment by llorllale** [comment]
@amihaiemil added new puzzle, fixed other puzzle, as per your suggestions

**Comment by llorllale** [comment]
@amihaiemil ping

**Comment by amihaiemil** [comment]
@llorllale Well, now appveyor seems to have some compilation errors and qulice complains in Travis. I know you said qulice errors were introduced in another PR and should be fixed, but what about compilation errors?

**Comment by llorllale** [comment]
@amihaiemil 
> qulice errors were introduced in another PR
PR for that is ready for review

> what about compilation errors?
Most likely because of the following in `appveyor.yml`:
```yml
environment:
  matrix:
    - JAVA_HOME: C:\Program Files\Java\jdk1.7.0
    #- JAVA_HOME: C:\Program Files\Java\jdk1.8.0
```

**Comment by llorllale** [comment]
@amihaiemil I'm not familiar with appveyor. I can try toggling one JAVA_HOME for the other. WDYT?

**Comment by amihaiemil** [comment]
@llorllale I see. Yeah, do it if you still have time. If not, open an issue :)

**Comment by llorllale** [comment]
@amihaiemil done.

Appveyor build now fails due to that "api rate limit exceeded" error

**Comment by 0crat** [comment]
@amihaiemil[/z](https://www.0crat.com/u/amihaiemil) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Pull request merged**